tests/rptest/tests/compatibility/sarama_test.py

```diff
@@ -0,0 +1,113 @@
+
+    # runs sarama unit tests
+    def test_sarama(self):
+        #Form the command
+        cmd = f"cd {SaramaTest.TESTS_DIR} && go test -count=1 ./..."
+
+        #Run the example
+        #Here, we're not using a BackgroundThreadServer (i.e., CompatExample)
+        #because the unit tests are run synchronously
+        node = self.redpanda.nodes[0]
+        result = node.account.ssh_output(cmd).decode().split('\n')
+
+        failed = False
+        i = 0
+        while not failed and i < len(result):
+            failed = 'FAIL' in result[i].rstrip()
+            i += 1
+
+        assert not failed, "sarama unit tests failed"
+
+    # run sarama interceptors example
+    def test_sarama_interceptors(self):
+        def event_found(line):
+            return 'SpanContext' in line
+
+        #Setup and run the example
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/interceptors")
+        cmd = f"interceptors -brokers {self.redpanda.brokers()} -topic {self.topic}"
+        cmds = (os.path.join(EXAMPLE_DIR, cmd), "interceptors")
+
+        self._example.set_example(cmds, event_found)
+        self._example.start()
+
+        #Let the example run for a few seconds
+        time.sleep(10)
+
+        assert self._example.ok(), "sarama interceptors test failed"
+
+    # Run sarama http server example
+    def test_sarama_http_server(self):
+        def server_running(line):
+            return 'Listening for requests' in line
+
+        #Setup and run the example
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/http_server")
+        cmd = f"http_server -addr n1:8080 -brokers n1:9092"
+        cmds = (os.path.join(EXAMPLE_DIR, cmd), "http_server")
+        node = self.redpanda.nodes[0]
+
+        self._example.set_example(cmds, server_running, node)
+        self._example.start()
+
+        #Let the example run for a few seconds
+        time.sleep(10)
+
+        assert self._example.ok(), "sarama http_server test failed to load"
+
+        #Requests on server
+        page = requests.get("http://n1:8080/")
+        self.logger.debug(f'sarama http page: {page.text}')
+
+        assert page.ok, f"sarama http_server test failed: {page.status_code}"
+
+    # run sarama consumergroup example
+    def test_sarama_consumergroup(self):
+        def msg_found(line):
+            return 'Message claimed:' in line
+
+        #Run the publisher
+        self._producer.start()
+
+        #Setup and run the example
+        EXAMPLE_DIR = os.path.join(SaramaTest.TESTS_DIR, "examples/consumergroup")
+        cmd = f"consumer -brokers=\"n1:9092\" -topics=\"{self.topic}\" -group=\"example\""
+        cmds = (os.path.join(EXAMPLE_DIR, cmd), "consumer")
+
+        self._example.set_example(cmds, msg_found)
+        self._example.start()
+
+        #Let the example run for a few seconds
+        time.sleep(10)
+
+        assert self._example.ok(), "sarama consumergroup test failed"
```

Comment:
you can configure whatever text editor you're using to automatically add newlines